stream_alert/rule_processor/threat_intel.py

```diff
@@ -251,24 +254,40 @@ def _process_ioc(self, ioc_collections):
             query_values = []
             for ioc in subset:
                 if ioc.value not in query_values:
-                    query_values.append(ioc.value.lower())
+                    query_values.append(ioc.value)
 
             query_result = []
 
-            result, unprocesed_keys = self._query(query_values)
-            query_result.extend(result)
+            query_error_msg = 'An error occured while quering dynamodb table. Error is: %s'
+            try:
+                result, unprocesed_keys = self._query(query_values)
+                query_result.extend(result)
+            except ClientError as err:
```

Comment:
why handle the different errors separately? you can catch both on a single line as a tuple:

`except (Error1, Error2) as err`

Reply:
It is because error message is different. `ClientError` has `response` attribute, `ParamValidationError` doesn't. It will be useful if we can log error details. So I separate them to be able to log details.

Reply:
Suggestion on this: use `LOGGER.exception` instead of `error` so we can see the full stack & exception value(s).